_posts/en/2024-02-27-kafka-aws-msk.md

```diff
@@ -0,0 +1,223 @@
+
+![Created Kafka cluster](/assets/img/posts/2024-02-27-kafka-aws-msk/kafka-aws-msk-2.png)
+<span class="img-legend">Created Kafka cluster</span>
+
+## Creating an IAM role
+To create an IAM policy, we need to go to the [IAM](https://console.aws.amazon.com/iam/home#/home) page and select **Policies** from the side menu, then **Create policy**.
+
+![Adding an authorization rule](/assets/img/posts/2024-02-27-kafka-aws-msk/kafka-aws-msk-3.png)
+<span class="img-legend">Adding an authorization rule</span>
+
+As for the editor, I suggest switching to the JSON version. It will make the configuration of policy easier. The following policy enables you to perform all operations on your Kafka cluster. Just remember to change the region, **Account-ID**, and **ClusterName**.
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "kafka-cluster:*"
+            ],
+            "Resource": [
+                "arn:aws:kafka:region:Account-ID:cluster/ClusterName/*"
+            ]
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "kafka-cluster:*"
+            ],
+            "Resource": [
+                "arn:aws:kafka:region:Account-ID:topic/ClusterName/*"
+            ]
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "kafka-cluster:*"
+            ],
+            "Resource": [
+                "arn:aws:kafka:region:Account-ID:group/ClusterName/*"
+            ]
+        }
+    ]
+}
+```
+Next, we need to attach the policy created above to the IAM role. To do this, select **Roles** from the side menu and then **Create roles**. Select **EC2** as the **Use case**, and on the next screen select the policy rule we created earlier. Click **Next**, name the role, and create the role using **Create roles**.
+
+After performing the above configuration, we have two options. The first is to run Kouncil on the AWS infrastructure, connect Kafka Amazon MSK, and set public access to Kouncil. The second option, in our opinion less secure, is to set public access to Kafka.
+
+## Option 1 - running Kouncil in AWS infrastructure
+To run Kouncil in an AWS infrastructure, we will need an Amazon EC2 instance. Create it [here](https://console.aws.amazon.com/ec2/) by clicking on **Launch instance**, and then name it. You can leave the default value for Amazon Machine Image (AMI). Choose **t2.micro** as the instance type. Then, if you don't have one, create a new login key pair and download it. After expanding **Advanced Details**, select in the **IAM instances profile** the IAM role created earlier. Then, click **Launch instance**.
+
+### Security group configuration
+To make Kouncil accessible from the outside, we need to configure the security group, specifically, allow access to the port on which Kouncil will be run. To do this, select the instance from the list and go to the **Security** tab.
```

Comment:
To zdanie:
"To make Kouncil accessible from the outside, we need to configure the security group, specifically, allow access to the port on which Kouncil will be run."
ma sens po Polsku, ale po przetłumaczeniu już nie :)

Reply:
Dla mnie to brzmi poprawnie w jednej i drugiej wersji